Frameworks/CoreText/CTFont.mm

```diff
@@ -811,8 +811,13 @@ void CTFontGetVerticalTranslationsForGlyphs(CTFontRef font, const CGGlyph glyphs
         }
 
         std::transform(metrics, metrics + count, translations, [font](const DWRITE_GLYPH_METRICS& metrics) {
-            return CGSize{ -0.5 * __CTFontScaleMetric(font, metrics.advanceWidth),
-                           -__CTFontScaleMetric(font, metrics.verticalOriginY - metrics.topSideBearing + sc_VerticalTranslationSpace) };
+            if (metrics.advanceWidth == metrics.rightSideBearing && metrics.verticalOriginY == metrics.topSideBearing) {
+                // Special case of empty glyph, should just return CGRectZero;
+                return CGSizeZero;
+            }
+
+            return CGSize{ -__CTFontScaleMetric(font, metrics.advanceWidth >> 1),
```

Comment:
it's common, but i also had to do a double-take and make sure that advanceWidth wasn't 1) a bit flag somehow 2) a signed int. imma double down and say you should just write what you mean here and let the compiler optimize.